### DISCUSSION TITLE
Implement table element

### DISCUSSION THREAD

**Discussion by adamberecz** [vueform/vueform#73] [Ideas]
Implement table element

**Comment by jadrake75** [comment]
oof.... just keep in mind tables and grids are the scourge of UI frameworks.  I evaluated about four before eventually just using ag-grid via their vue wrapper.  I could see the value of the VueformBuilder offering the ability to build and including a grid.  Perhaps create a plugin interface that you can then have users install a grid plugin and expose it that way.  Something like ag-grid-for-vueform module or something and then abstract the grid portions generically.  I have seen tens of Millions invested by companies and huge teams in developing tables and grids..... Just be careful it doesn't suck you in!

I did this in my project with a DataGridComponent that wraps a ag-grid (I could swap it out) and provides a simple properties map of rowData, columnDefs (which is my own model but largely mirrors ag-grid model), selectionData, etc. and it emits some simple events like selected, deselected and sortChanged.  Any ag-grid stuff is entirely in that class only (even though I use my grid in ~10 other places) 

While this is specific to my app in theory the approach is adaptable: https://github.com/stamp-web/stamp-web-vuejs/blob/main/src/components/table/DataGridComponent.vue

**Reply by adamberecz** [reply]
I love your wrapper and this is also how I would imagine including a grid style table element like ag-grid in Vueform. Thanks for sharing your thoughts and the example. Developing our own grid solution is probably out of question at this point, if not something minimal with only cols, rows and pagination.

**Comment by nclemeur** [comment]
I think it would be better to add a bit more details about a 'Table' element. In the Vueform context, I would imagine this is more like a layout element that helps position the components on a form? Is that what you are envisaging?
I find the columns system not flexible enough for my need, but I did implement a CSS grid based system (I wish I could completely disable the columns for that purpose as I had to do some ugly CSS overrides).

**Reply by adamberecz** [reply]
I agree that this needs clarification, because "table" could mean at least 4 different things, as people here mention some of them:
- table layout for organizing elements
- a matrix of radio / checkbox options
- data table with cols, rows and pagination (native or plugin based like ag-grid)
- spreadsheet like element (external lib)

Probably I would follow this order.

**Comment by adamberecz** [comment]
Finally 🎉
https://vueform.com/reference/matrix-element

The quickest way to check out its features could be to search for "Matrix" or "Table" element in the builder:
https://builder.vueform.com/demo

**Comment by adamberecz** [comment]
Another element that could be related to this is out as well with `1.12.0`:
https://vueform.com/reference/grid-element#gridelement
